pkg/vsphere/tags/rest_client.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tags
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/juju/errors"
+)
+
+const (
+	RestPrefix = "/rest"
+)
+
+type RestClient struct {
+	host     string
+	scheme   string
+	endpoint *url.URL
+	HTTP     *http.Client
+	cookies  []*http.Cookie
+}
+
+func NewClient(u *url.URL, insecure bool, thumbprint string) *RestClient {
+	log.Debugf("Create rest client")
+	u.Path = RestPrefix
+
+	if thumbprint != "" {
+		thumbprint = strings.Replace(thumbprint, ":", "", -1)
+	}
+
+	// #nosec
+	c := &RestClient{
+		endpoint: u,
+		host:     u.Host,
+		scheme:   u.Scheme,
+		HTTP: &http.Client{
+			Transport: &http.Transport{
+				DialTLS: func(network, addr string) (net.Conn, error) {
+					c, err := tls.Dial(network, addr, &tls.Config{InsecureSkipVerify: insecure})
+					if err == nil {
+						return c, nil
+					}
+
+					switch err := err.(type) {
```

Comment:
comment about what this accomplishes - I'm too used to C style fallthrough for this to be obvious without thought.